src/elements/content-explorer/ContentExplorer.js

```diff
@@ -391,12 +399,24 @@ class ContentExplorer extends Component<Props, State> {
      * @param {Boolean|void} triggerNavigationEvent - To trigger navigate event and focus grid
      * @return {void}
      */
-    fetchFolderSuccessCallback(collection: Collection, triggerNavigationEvent: boolean): void {
-        const { onNavigate, rootFolderId }: Props = this.props;
-        const { id, name, boxItem }: Collection = collection;
+    async fetchFolderSuccessCallback(collection: Collection, triggerNavigationEvent: boolean): Promise<void> {
+        const { features, onNavigate, rootFolderId }: Props = this.props;
+        const { boxItem, id, name }: Collection = collection;
         const { selected }: State = this.state;
         const rootName = id === rootFolderId ? name : '';
 
+        if (isFeatureEnabled(features, 'contentExplorer.gridView.enabled')) {
+            const { items = [] } = collection;
+            const fileAPI = this.api.getFileAPI();
+            const thumbnailUrlPromises = items.map(item => fileAPI.getThumbnailUrl(item));
+
+            await Promise.all(thumbnailUrlPromises).then(thumbnailUrls => {
+                collection.items = items.map((item, index) => {
```

Comment:
We lose a lot of the benefit of async/await and Promises by mixing them; refactoring to use one or the other should remove the need to modify `collection` directly. Here's one potential solution using await/async:

```
const { items } = collection;
const itemThumbnails = await Promise.all(thumbnailUrlPromises);
const itemsWithThumbnails = items.map((item, index) => {
    const thumbnailUrl = thumbnailUrls[index];
    return thumbnailUrl ? { ...cloneDeep(item), thumbnailUrl } : item;
});
this.updateCollection({ ...collection, items: itemsWithThumbnails }, selected);
```